identity/webapp/src/frontend/hooks/useUserInfo.ts

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+
+export type UserInfo = {
+  firstName: string;
+  lastName: string;
+  email: string;
+  barcode: string;
+};
+
+type UserInfoQuery = {
+  data?: UserInfo;
+  isLoading: boolean;
+  error?: string;
+};
+
+export function useUserInfo(): UserInfoQuery {
+  const [data, setData] = useState<UserInfo>();
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      await axios
+        .get<UserInfo>('/api/users/me')
```

Comment:
`me` in the URL should be a `user_id`, but the endpoint doesn't need one. See discussion above.